### PR TITLE
ci: added test coverage reporting.

### LINKED ISSUE #53: [Bounty] add coverage reporting and add to github actions
# Background 
Currently the SDK doesn't have code coverage reports being generated that makes it bit hard to bounty out more tests or see where we are missing tests easily. 

# Acceptance Criteria
- yarn coverage will generate a coverage report based on our unit tests
- coverage stats  / changes are shown for PRs in github

# Bounty Payout
ElasticSwap will compensate a bounty hunter 40 TIC to set this up

### PR DESCRIPTION
**Pull request by zhyd1997**
Resolves #53 

@0xean You need to integrate this repo with [Codecov](https://about.codecov.io/) and add `CODECOV_TOKEN` to the repo secretes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # elasticswap-sdk
 The official SDK for ElasticSwap protocol and token contracts.
 
+[![codecov](https://codecov.io/gh/zhyd1997/elasticswap-sdk/branch/development/graph/badge.svg?token=Q6MSEE6HUN)](https://codecov.io/gh/zhyd1997/elasticswap-sdk)
```
Comment (zhyd1997) [inline README.md:4]: @0xean pls replace the line of code with the correct repo badge after you integrated the repo with Codecov:

codecov badge

**Review by zhyd1997 — COMMENTED** [review]
Request changes

**Comment by zhyd1997** [comment]
**Test Coverage** (`yarn coverage`) :

test coverage

File: README.md (hunk 1 above)
Comment (0xean) [reply README.md:4]: [codecov](https://codecov.io/gh/ElasticSwap/elasticswap-sdk)

File: nyc.config.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
```
Comment (0xean) [inline nyc.config.js:1]: Do we also need this nyc config?  Seems like everything runs fine without it, please remove.

**Review by 0xean — COMMENTED** [review]
Left the link for the badge, and a small comment re: extra config file for nyc.

File: nyc.config.js (hunk 2 above)
Comment (zhyd1997) [reply nyc.config.js:1]: Sorry, forgot it.

File: README.md (hunk 1 above)
Comment (zhyd1997) [reply README.md:4]: Yes, because we never add code coverage before, so it's unknown.

**Pull request merged**